pandas/tests/indexes/test_category.py

```diff
@@ -540,6 +540,24 @@ def test_reindex_dtype(self):
         tm.assert_numpy_array_equal(indexer,
                                     np.array([0, 3, 2], dtype=np.intp))
 
+    def test_reindex_duplicate_target(self):
+        # See GH23963
+        c = CategoricalIndex(['a', 'b', 'c', 'a'],
+                             categories=['a', 'b', 'c', 'd'])
+        with pytest.raises(ValueError, match='non-unique indexer'):
+            c.reindex(['a', 'a', 'c'])
+
+        with pytest.raises(ValueError, match='non-unique indexer'):
+            c.reindex(CategoricalIndex(['a', 'a', 'c'],
+                                       categories=['a', 'b', 'c', 'd']))
+
+        # See GH21809
```

Comment:
can you put this tests in pandas/test/frame/test_analytics (just the part about round)

Reply:
Tests are passing except for a lint error in master - have a fix at https://github.com/pandas-dev/pandas/pull/23989